### ISSUE TITLE
Improving SC MP2RAGE manual segmentations

### ISSUE THREAD

**Issue by Nilser3** [neuropoly/data-management#268]
# Description 

It has been observed that manual segmentations of SC in MP2RAGE datas (`basel-mp2rage` `marseille-3T-mp2rage`) have some issues like:

- #266 
- SC masks does not cover the entire SC

![image](https://github.com/neuropoly/data-management/assets/77469192/1c8eb9c8-cfba-4a57-8f9f-12caa0cb1323)

This issue is for improve these SC segmentations
Here the first QC in some MP2RAGE subjects: https://amubox.univ-amu.fr/s/FBAfYqcGwGXRGRy 

# Related issues
#266 #267

**Comment by jcohenadad** [comment]
@Nilser3 as discussed and agreed on (https://github.com/neuropoly/data-management/issues/266#issuecomment-1743810312) the plan was to apply the contrast-agnostic model on these data, and _then_ manually correct the output segmentations. Was there a misunderstanding somewhere? 

Also, when sharing the QC, pls also add the current segmentation so we can compare with your corrections.

**Comment by Nilser3** [comment]
understood,
I attach the QC [here ](https://amubox.univ-amu.fr/s/qZBqCnT7ynSTYLa), where we can see:
- GT :  __label-SC_seg.nii.gz_
- contrast-agnostic model (binary thr 0.5) : __pred_bin.nii.gz_
- Nilser correction: __rater2.nii.gz_

**Comment by jcohenadad** [comment]
@Nilser3 based on what I am seeing, you corrected "_label-SC_seg.nii.gz", not "_pred_bin.nii.gz". This is different than what is described in https://github.com/neuropoly/data-management/issues/268#issuecomment-1751146033.

**Comment by Nilser3** [comment]
okay,
To do the manual correction I used both masks (GT and pred_bin),
The main difference is the starting point of the SC (image attached)
![image](https://github.com/neuropoly/data-management/assets/77469192/b7ab9b26-97d9-474d-ad66-98dd1b3a4bc1)

So I'll go back to making corrections based on the "_pred_bin.nii.gz" alone.

**Comment by jcohenadad** [comment]
> To do the manual correction I used both masks (GT and pred_bin)

What do you mean by "I used both"? Did you mean visually, or algorithmically (eg: summation of both masks), or other? Please elaborate.

When flipping back-and-forth between the GT and _rater2, and then between pred_bin and _rater2, I notice that in most slices, the contour of the SC on _rater2 is more similar to GT than pred_bin. Given that the goal of the contrast-agnostic project is to avoid propagating the bias of the previous algo (deepseg_sc, which was applied to GT), we need to be very careful with choosing the starting point of the SC segmentation for manual correction. Tagging @sandrinebedard @naga-karthik @valosekj @plbenveniste who can further clarify if something is unclear in my explanation

**Comment by Nilser3** [comment]
> What do you mean by "I used both"? Did you mean visually, or algorithmically (eg: summation of both masks), or other? Please elaborate.
> 

Sorry, I was not clear,
Indeed, I did a sum of the masks (GT and pred_bin), then a binarization of the result and finally I manually corrected this new mask

> When flipping back-and-forth between the GT and _rater2, and then between pred_bin and _rater2, I notice that in most slices, the contour of the SC on _rater2 is more similar to GT than pred_bin

In general, the pred_bin masks seem eroded when compared to the GT, perhaps that is why the effect is seen that the corrections are more similar to the GT

**Comment by jcohenadad** [comment]
> In general, the pred_bin masks seem eroded when compared to the GT, perhaps that is why the effect is seen that the corrections are more similar to the GT

Exactly. And that is precisely the issue. By summing the two SC segmentations, you will keep the bigger one, which might not be the 'most accurate' one.

**Comment by Nilser3** [comment]
Thanks you! Now it makes more sense to me, 
So I will proceed to redo the segmentations but based only on the _pred_bin_ masks

**Comment by Nilser3** [comment]
[Here ](https://amubox.univ-amu.fr/s/Ness82qmWJgbQdW) is a manual correction, based only on the binarized images (pred_bin) of the contrast-agnostic masks (thr = 0.5001).

**Comment by jcohenadad** [comment]
> [Here ](https://amubox.univ-amu.fr/s/Ness82qmWJgbQdW)is a manual correction, based only on the binarized images (pred_bin) of the contrast-agnostic masks (thr = 0.5001).

You did a great job, Nilser! My only concern is that your manual correction seems to 'over-segment' compared to what the contrast-agnostic model produces. For example:
- contrast-agnostic: 
![image](https://github.com/neuropoly/data-management/assets/2482071/4e3c0ed6-a554-462b-a653-f625b2562bf9)
- your correction:
![image](https://github.com/neuropoly/data-management/assets/2482071/a7c6e84b-dc8c-4609-a7a2-e6a203d7ed66)

I would suggest to not alter _too much_ the outer boundaries of the contrast-agnostic model (at the risk of introducing a bias when re-training the model with active learning), and primarily focus on:
- Fixing obvious under-segmentation caused by a lesion or artifact
  ![image](https://github.com/neuropoly/data-management/assets/2482071/a7dd55d8-e544-4062-b7eb-1f62dc857f7f)
- Fixing obvious missing segmentation at the edge of a volume:
  ![image](https://github.com/neuropoly/data-management/assets/2482071/51380bef-f992-4325-b784-f65a0eeddac5)

Tagging @sandrinebedard @naga-karthik @valosekj @plbenveniste in case they have additional feedback

Also, feel free to upload the ZIP directly in this issue, in case the AMU link breaks in the future

**Comment by jcohenadad** [comment]
Proposed strategy: https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/issues/84

**Comment by jcohenadad** [comment]
Another more recent QC: https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/issues/84#issuecomment-1816557099

**Comment by Nilser3** [comment]
Here is a [QC](https://amubox.univ-amu.fr/s/qNiTcKdXoC3gMHN) of manual correction on the entire `marseille-3T-mp2rage` dataset , based only on the binarized contrast-agnostic masks

**Legend of QC maks**

- SC_seg.nii.gz -> sct_deepseg_sc, corrected by rater 1 (Samira)
- SC_seg_rater2.nii.gz -> contrast-agnostic ([version v2.0](https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/releases/tag/v2.0)), binarized with threshold 0.5001 corrected by rater 2 (Nilser)  + union with MS lesion masks (label-lesion_rater2.nii.gz)

**Comment by jcohenadad** [comment]
I suggest we put a hold on the correction of the masks until this issue is settled: https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/issues/99
